image/image_test.go

```diff
@@ -516,55 +541,64 @@ func (s *imageSuite) TestSetupSeed(c *C) {
 	err := image.SetupSeed(s.tsto, s.model, opts)
 	c.Assert(err, IsNil)
 
-	// check seed yaml
-	seedYaml, err := seed.ReadYaml(filepath.Join(seeddir, "seed.yaml"))
-	c.Assert(err, IsNil)
-
-	c.Check(seedYaml.Snaps, HasLen, 4)
+	// check seed
+	seeddir := filepath.Join(rootdir, "var/lib/snapd/seed")
+	seedsnapsdir := filepath.Join(seeddir, "snaps")
+	essSnaps, runSnaps, roDB := s.loadSeed(c, seeddir)
+	c.Check(essSnaps, HasLen, 3)
+	c.Check(runSnaps, HasLen, 1)
 
 	// check the files are in place
 	for i, name := range []string{"core", "pc-kernel", "pc"} {
 		info := s.AssertedSnapInfo(name)
 		fn := filepath.Base(info.MountFile())
 		p := filepath.Join(seedsnapsdir, fn)
-		c.Check(osutil.FileExists(p), Equals, true)
+		c.Check(p, testutil.FilePresent)
+		c.Check(essSnaps[i], DeepEquals, &seed.Snap{
+			Path: p,
+
+			SideInfo: &info.SideInfo,
+
+			Essential: true,
+			Required:  true,
 
-		c.Check(seedYaml.Snaps[i], DeepEquals, &seed.Snap16{
-			Name:    name,
-			SnapID:  s.AssertedSnapID(name),
 			Channel: stableChannel,
-			File:    fn,
 		})
 		// sanity
 		if name == "core" {
-			c.Check(seedYaml.Snaps[i].SnapID, Equals, "coreidididididididididididididid")
+			c.Check(essSnaps[i].SideInfo.SnapID, Equals, "coreidididididididididididididid")
 		}
 	}
-	c.Check(seedYaml.Snaps[3].Name, Equals, "required-snap1")
-	c.Check(seedYaml.Snaps[3].Contact, Equals, "foo@example.com")
+	c.Check(runSnaps[0], DeepEquals, &seed.Snap{
+		Path: filepath.Join(seedsnapsdir, filepath.Base(s.AssertedSnapInfo("required-snap1").MountFile())),
 
-	storeAccountKey := s.StoreSigning.StoreAccountKey("")
-	brandPubKey := s.Brands.PublicKey("my-brand")
+		SideInfo: &s.AssertedSnapInfo("required-snap1").SideInfo,
 
-	// check the assertions are in place
-	for _, fn := range []string{"model", brandPubKey.ID() + ".account-key", "my-brand.account", storeAccountKey.PublicKeyID() + ".account-key"} {
-		p := filepath.Join(seedassertsdir, fn)
-		c.Check(osutil.FileExists(p), Equals, true)
```

Comment:
Is this test still checking (indirectly) that the files are on disk? Or is this now the wrong layer for this kind of test?